include/pybind11/functional.h

```diff
@@ -50,9 +50,9 @@ struct type_caster<std::function<Return(Args...)>> {
         }
 
         auto src = reinterpret_borrow<object>(src_);
-        value = [src](Args... args) -> Return {
+        value = [src](Args&&... args) -> Return {
```

Comment:
The `std::forward` below is a nice catch, but this `&&` is not needed here. The `Args` types have been deduced prior to this line, so the `&&` will not have the usual forwarding reference behavior.

```c++
template<class T> 
void foo(T&& x) {} // forwarding reference

template<class T>
struct A {
    static void bar(T&& x) {} // && added to existing type
};

auto value = 1;
foo(value); // fine: T == int& --> foo(int& x)
A<int>::bar(value); // error: T == int --> bar(int&& x) <-- rvalue reference can't bind to lvalue
```